### PR TITLE
cmake: make musl support distro agnostic

### PR DESCRIPTION
**Pull request by abenson**

### CODE REVIEW

**Comment by abenson** [comment]
[Recommended here](https://github.com/void-linux/void-packages/pull/20798#discussion_r435907617) to better improve support for `musl` based distros.

File: CMakeLists.txt [hunk 1]
```diff
@@ -334,18 +334,11 @@ include_directories(BEFORE ${OPENSSL_INCLUDE_DIR})
...
-        RESULT_VARIABLE os_release_alpine
-    )
-
-    if ( os_release_alpine EQUAL 0 )
-        find_package(FTS REQUIRED)
-        list(APPEND OPTLIBS ${FTS_LIBRARY})
-        include_directories(BEFORE ${FTS_INCLUDE_DIR})
-    endif ()
+# Musl support
+if ( NEEDS_LIBFTS )
```
Comment (timwoj) [inline CMakeLists.txt:338]: Where is this variable defined? I see it defined in `srcpkgs/zeek/template` in the other project, but nowhere in Zeek.

**Comment by abenson** [comment]
It would need to be defined at build. As is, zeek only checks for Alpine
and not anything musl.

If Alpine does end up shipping Zeek (checked to link there as well) then
they just have to set the same variable in the APKBUILD.

**Comment by 0xxon** [comment]
Is there a way to change this so that cmake can itself figure out if this is required? I don't like the thought of relying on a magic variable being set for this to work.

**Comment by timwoj** [comment]
> Is there a way to change this so that cmake can itself figure out if this is required? I don't like the thought of relying on a magic variable being set for this to work.

That was my thought as well. This change also breaks an already working setup for Alpine users in favor of a variable that they need to know they have to set.

**Comment by abenson** [comment]
This patch seems to be better. I'm still testing to make sure the compile test doesn't break cross-compilation.

**Pull request merged**